include/CppUTest/CppUTestConfig.h

```diff
@@ -211,13 +213,31 @@ typedef unsigned long long cpputest_ulonglong;
 /* Define some placeholders to disable the overloaded methods.
  * It's not required to have these match the size of the "real" type, but it's occasionally convenient.
  */
+
 #if defined(CPPUTEST_64BIT) && !defined(CPPUTEST_64BIT_32BIT_LONGS)
-typedef struct { char dummy[16]; } cpputest_longlong;
-typedef struct { char dummy[16]; } cpputest_ulonglong;
+#define CPPUTEST_SIZE_OF_FAKE_LONG_LONG_TYPE 16
 #else
-typedef struct { char dummy[8]; } cpputest_longlong;
-typedef struct { char dummy[8]; } cpputest_ulonglong;
+#define CPPUTEST_SIZE_OF_FAKE_LONG_LONG_TYPE 8
 #endif
+
+struct cpputest_longlong
+{
+#if defined(__cplusplus)
+  cpputest_longlong() {}
+  cpputest_longlong(int) {}
+#endif
+  char dummy[CPPUTEST_SIZE_OF_FAKE_LONG_LONG_TYPE];
+};
+
+struct cpputest_ulonglong
+{
+#if defined(__cplusplus)
+  cpputest_ulonglong() {}
+  cpputest_ulonglong(int) {}
+#endif
+  char dummy[CPPUTEST_SIZE_OF_FAKE_LONG_LONG_TYPE];
+};
+
```

Comment:
These won't work with C code -- they have to be typedefs to avoid syntax changes.  eg:

    typedef struct cpputest_longlong_
    {
    #if defined(__cplusplus)
        cpputest_longlong_() {}
        cpputest_longlong_(int) {}
    #endif
        char dummy[CPPUTEST_SIZE_OF_FAKE_LONG_LONG_TYPE];
    } cpputest_longlong;

Although I'm still unconvinced of the merit of including an int constructor.  It won't prevent real tests from failing with compile errors (since the only tests that would compile are ones that use small integer constants, and other than in CppUTest's own tests that will never happen).  So all it really gains is the ability to use `0` and `1` in CppUTest's own tests instead of declaring a dummy `cpputest_longlong` default-constructed variable as done previously (via the macro in my version, or on a separate line in yours).